mobile/examples/object_detections/iOS/ORTObjectDetection/ModelHandler.swift

```diff
@@ -0,0 +1,463 @@
+        }
+
+        return labelData
+    }
+    
+    private func colorForClass(withIndex index: Int) -> UIColor {
+        
+        // Assign variations to the base colors for each object based on its index.
+        let baseColor = colors[index % colors.count]
+        
+        var colorToAssign = baseColor
+        
+        let percentage = CGFloat((10 / 2 - index / colors.count) * 10)
+        
+        if let modifiedColor = baseColor.getModified(byPercentage: percentage) {
+            colorToAssign = modifiedColor
+        }
+        
+        return colorToAssign
+    }
+    
+    // Return the RGB data representation of the given image buffer.
+    func rgbDataFromBuffer(
+        _ buffer: CVPixelBuffer,
+        byteCount: Int,
+        isModelQuantized: Bool = false
+    ) -> Data? {
+
+        CVPixelBufferLockBaseAddress(buffer, .readOnly)
+        defer {
+            CVPixelBufferUnlockBaseAddress(buffer, .readOnly)
+        }
+        guard let sourceData = CVPixelBufferGetBaseAddress(buffer) else {
+            return nil
+        }
+        
+        let width = CVPixelBufferGetWidth(buffer)
+        let height = CVPixelBufferGetHeight(buffer)
+        let sourceBytesPerRow = CVPixelBufferGetBytesPerRow(buffer)
+        let destinationChannelCount = 3
+        let destinationBytesPerRow = destinationChannelCount * width
+        
+        var sourceBuffer = vImage_Buffer(data: sourceData,
+                                         height: vImagePixelCount(height),
+                                         width: vImagePixelCount(width),
+                                         rowBytes: sourceBytesPerRow)
+        
+        guard let destinationData = malloc(height * destinationBytesPerRow) else {
+            print("Error: out of memory")
+            return nil
+        }
+        
+        defer {
+            free(destinationData)
+        }
+        
+        var destinationBuffer = vImage_Buffer(data: destinationData,
+                                              height: vImagePixelCount(height),
+                                              width: vImagePixelCount(width),
+                                              rowBytes: destinationBytesPerRow)
+        
+        let pixelBufferFormat = CVPixelBufferGetPixelFormatType(buffer)
+        
+        switch (pixelBufferFormat) {
+        case kCVPixelFormatType_32BGRA:
+            vImageConvert_BGRA8888toRGB888(&sourceBuffer, &destinationBuffer, UInt32(kvImageNoFlags))
+        case kCVPixelFormatType_32ARGB:
+            vImageConvert_ARGB8888toRGB888(&sourceBuffer, &destinationBuffer, UInt32(kvImageNoFlags))
+        case kCVPixelFormatType_32RGBA:
+            vImageConvert_RGBA8888toRGB888(&sourceBuffer, &destinationBuffer, UInt32(kvImageNoFlags))
+        default:
+            // Unknown pixel format.
+            return nil
+        }
+        
+        let byteData = Data(bytes: destinationBuffer.data, count: destinationBuffer.rowBytes * height)
```

Comment:
looks like there are some copies, e.g. destinationBuffer -> byteData and floats -> return value. is it possible to remove some of these?